backend/src/packages/users/user.repository.ts

```diff
@@ -163,12 +163,24 @@ class UserRepository implements IRepository {
   }
 
   public async getAllAuthors(): Promise<UserDetailsModel[] | null> {
-    const authors = await UserDetailsModel.query().distinctOn('userId')
-      .whereRaw(`
-      EXISTS(SELECT 1
-      FROM articles,user_details
-      WHERE articles.user_id = user_details.user_id)
-    `);
+    const authors = await UserDetailsModel.query()
+      .from((builder) => {
+        void builder
+          .distinctOn('userId')
+          .select('userId', 'firstName', 'lastName')
+          .from('user_details')
```

Comment:
Are you sure we need to have separated .from statement as it's already in UserDetailsModel

Reply:
If I remove separated .from and leave the code like so
```
const authors = await UserDetailsModel.query()
      .distinctOn('userId')
      .select('userId', 'firstName', 'lastName')
      .from('user_details')
      .whereRaw(
        `
        EXISTS(SELECT 1
        FROM articles,user_details
        WHERE articles.user_id = user_details.user_id)
      `,
      )
      .orderBy('userId')
      .as('subquery')
      .orderBy('firstName')
      .orderBy('lastName');
```
the sorting doesn't work. Did I miss something?

Reply:
Not sure that aliasing is needed, and I believe at first you should order by firstname\lastname,, and only after that with userId

Reply:
And let's have sorting on users with `orderBy('CONCAT(first_name, ' ', last_name)')` if it's possible

Reply:
As I have checked, the expressions specified in the DISTINCT ON clause must also appear in the ORDER BY clause in the same order.
So, it's not possible to order by firstname\lastname at first. That's why, I used a subquery here.

Reply:
Okay